gobblin-data-management/src/main/java/gobblin/data/management/copy/RecursiveCopyableDataset.java

```diff
@@ -44,20 +44,14 @@
 
   private final Path rootPath;
   private final FileSystem fs;
-  private final Path targetDirectory;
   private final Properties properties;
   private LoadingCache<Path, OwnerAndPermission> ownerAndPermissionCache;
   private final PathFilter pathFilter;
 
   public RecursiveCopyableDataset(final FileSystem fs, Path rootPath, Properties properties) {
 
-    Preconditions.checkArgument(
```

Comment:
It is no longer the responsibility of the Dataset to figure out the target, so it doesn't need this option anymore.